tensorboard/plugins/debugger/README.md

```diff
@@ -0,0 +1,144 @@
+
+* enables users to pause execution at specified ops or continue execution for some number of steps.
+* visualizes values of tensors.
+* associates tensors with specific lines in python code.
+
+![The Debugger Dashboard](images/debugger_intro.png)
+
+# Setup
+
+## Instrumenting Model Logic
+
+To use this dashboard, construct a `TensorBoardDebugWrapperSession`. Subsequently, that wrapper session will issue [gRPC](https://grpc.io/docs/guides/) messages to TensorBoard that containing data for debugging.
+
+The constructor accepts these parameters.
+
+1. the original `tf.Session` object.
+2. the `[[host]]:[[port]]` address to which to stream gRPC messages.
+
+Example logic:
+
+```python
+from tensorflow.python import debug as tf_debug
+sess = tf.Session()
+sess = tf_debug.TensorBoardDebugWrapperSession(sess, 'localhost:6064')
+sess.run(my_fetches)
+```
+
+## Start TensorBoard with the `debugger_port` flag.
+
+In addition to instrumenting the model, the user must pass the `debugger_port` flag to TensorBoard. This port number matches the one above. TensorBoard will then both receive gRPC messages from and issue gRPC messages to model logic.
+
+This example command demonstrates how to start TensorBoard:
+
+```
+tensorboard \
+    --logdir ~/project_foo/model_bar_logdir \
+    --port 6006 \
+    --debugger_port 6064 \
+;
+```
+
+## Navigating to TensorBoard
+
+Navigate to the debugger dashboard within TensorBoard based on which port it serves on (specified via the `port` flag). For instance, the URL might be `http://localhost:6006#debugger`.
+
+Initially, a dialog may indicate that the dashboard is waiting on a session run to begin. The dialog will hide once the latter happens.
+
+![Dialog on waiting for the first session run](images/debugger_dialog.png)
+
+# Selecting Ops
+
+To examine the output value of an op, ops in the graph must be first selected. Subsequently, the debugger dashboard will pause runs at those ops, enabling users to examine outputs of those ops.
+
+## The Node List
+
+Ops can be selected via the the node list on the top left:
+
+![Node list](images/node_list.png)
+
+Toggling a checkbox next to an entire scope selects or deselects all ops under the scope. The checkbox for a scope is orange if some but not all of the ops within it are selected.
+
+The ops shown in the list may be filtered by regex based on name or type. Afterwards, ops may be more efficiently selected by the user.
+
+![Select ops by regex](images/regex_node_list.png)
+
+Clicking the link next to an op makes the graph explorer pan to it (and expand nodes if need be to show the op):
+
+![Make the graph explorer pan to a node](images/node_linking.png)
+
+## The Graph Explorer
+
+The graph explorer (on the right side of the dashboard) offers another way to select an op: A context menu appears when an op is right-clicked.
```

Comment:
This is a somewhat obscure feature. So I suggest  bold-face this word to bring more attention to it.